### PR TITLE
Update Apple OAuth docs, recommend apple-signin-auth for new apple integrations

### LINKED ISSUE #181: Apple OAuth2 flow bugged
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.8.1

### Plugin version

6.1.0

### Node.js version

16.14

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.3.1

### Description

Hello,

I am facing this issue since previous version of @fastify/oauth2: the authorizationMethod for Apple OAuth2 login is wrongly set to header, leading any login attempt to fail, instead the good format is classic form-body.

What I could do until the previous versions of the plugin was the following workaround code, that I was executing right after the plugin registration:

```
type OAuth2ReservedNamespace = {
      oauth2: {
         authorizationCode: {
            config: {
               options: {
                  authorizationMethod?: string;
               };
            };
         };
      };
   };

   // We need to delete Auth method, cause it has been wrongly set to Header in the library, but apple uses form-body
   delete (fastify.apple as unknown as OAuth2ReservedNamespace).oauth2
      .authorizationCode.config.options.authorizationMethod;
```
      
With this piece of code I was force removing the header auth method, allowing the auth flow to work.

But in the last release you have changed the config field to class private field (#config) and now I can't modify it even with the workaround.

Since this problem is becoming quite annoying, can you please change the #config.options.authorizationMethod value for Apple from your repository so that it's set to form-body, or is unset since it should be the default behavior.

Thanks in advance

### Steps to Reproduce

Just put in place an Apple OAuth2 flow with the plugin and try to login with Apple ID

### Expected Behavior

Usage of form body for authorization

**UPDATE: Ok I've finally found the solution:**

      credentials: {
         client: {
            id: ...,
            secret: ...,
         },
         **options: {
            authorizationMethod: 'body',
         },**
         auth: oauthPlugin.APPLE_CONFIGURATION,
      },

The options section could be added in th credentials section of the plugin registration option.

A bit tricky to understand. It would be nice a small notion about this at least in the github page, to avoid devs getting mad.

Especially valid for Apple ID that has this special configuration required.

**UPDATE 2**: Some interfaces still need to be adjusted ad hoc for Apple login. For example the OAuth2Token is missing some extra field outputted by the Apple login process.
In particular OAuth2Token.token.id_token is a missing field, needed to extract user email. 
I do realize that is a special behavior only belonging to Apple, but a specialized version of the token may be good for users to be aware again.

Thanks again

### PR DESCRIPTION
**Pull request by avmaxim**
#### Description
This PR is emerged due to the specifics of Apple OAuth2 authorization flow.
It provides a thorough explanation and example of a proper Apple OAuth2 usage.

Related issues:
* https://github.com/fastify/fastify-oauth2/issues/181 - can be closed due to this PR
* https://github.com/fastify/fastify-oauth2/issues/116

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by avmaxim** [comment]
@climba03003 @mcollina @allevo Hello guys! A bit of contribution here for Apple OAuth2 specifics. Please review and let me know if there's anything to add before merging to master. Thanks!

**Comment by Uzlopak** [comment]
Lets run the ci

**Comment by playleaks** [comment]
All good just a small linting problem

File: examples/apple.js [hunk 1]
```diff
@@ -1,61 +1,130 @@
...
+      secret: CLIENT_SECRET
     },
-    auth: oauthPlugin.APPLE_CONFIGURATION
+    auth: oauthPlugin.APPLE_CONFIGURATION,
+    options: {
+      /**
+       * Based on offical Apple OAuth2 docs, an HTTP POST request is sent to the redirectURI for the `form_post` value.
+       * And the result of the authorization is stored in the body as application/x-www-form-urlencoded content type.
+       * See {@link https://developer.apple.com/documentation/sign_in_with_apple/request_an_authorization_to_the_sign_in_with_apple_server}
+       */
```
Comment (Uzlopak) [inline examples/apple.js:36]:
```suggestion
       */
```

**Comment by avmaxim** [comment]
@Uzlopak @playleaks thx for a review!

@climba03003 @mcollina @allevo may you merge it into master? All CI checks and code review have passed ✅

**Review by climba03003 — COMMENTED** [review]
LGTM.

I think we should refine how the API should be for this plugin.

**Review by climba03003 — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by avmaxim** [comment]
@mcollina @climba03003  If it looks good, may you merge it then please? I along with other team mates want to use it in our project. I don't have rights to merge. Thanks!

**Comment by climba03003** [comment]
>  If it looks good, may you merge it then please?

Yes, but this PR doesn't block you from using this plugin. All the changes just `example` and `document`.
No actual code changes is made.

**Pull request merged**